dali/sasha/saarch.cpp

```diff
@@ -85,213 +85,343 @@ static void mkDateCompare(bool dfu,const char *dt,StringBuffer &out,char fill)
+            }
+        }
+
+    public:
+        IMPLEMENT_IINTERFACE_USING(CInterface);
+
+        CWUReader(ISashaCommand *cmd, const char *_mask)
+        {
+            mask.set(_mask);
+            owner.set(cmd->queryOwner());
+            cluster.set(cmd->queryCluster());
+            jobName.set(cmd->queryJobName());
+            state.set(cmd->queryState());
+            priority.set(cmd->queryPriority());
+            eclContains.set(cmd->queryEclContains());
+            fileRead.set(cmd->queryFileRead());
+            fileWritten.set(cmd->queryFileWritten());
+            cmdName.set(cmd->queryDfuCmdName());
+
+            dfu = cmd->getDFU();
+            hasWUSOutput = cmd->getAction()==SCA_WORKUNIT_SERVICES_GET;
+            outputModifiedTime = cmd->getAction()==SCA_LISTDT;
+            outputXML = cmd->getAction()==SCA_GET;
+
+            //In WUiterate(), the queryAfter() is used as fromDT and the queryBefore() is used as toDT.
+            mkDateCompare(dfu, cmd->queryAfter(), fromDT, '0');
+            mkDateCompare(dfu, cmd->queryBefore(), toDT, '9');
+        }
+        void getWUs(ISashaCommand *cmd, const char *beforeWU, const char *afterWU, unsigned numWUs)
+        {
+            StringBuffer path;
+            if (dfu)
+                getLdsPath("Archive/DFUWorkUnits", path);
+            else
+                getLdsPath("Archive/WorkUnits", path);
+            Owned<IFile> dir = createIFile(path.str());
+
+            StringArray outputFields;
+            const char *outputFormat = cmd->queryOutputFormat();
+            outputFields.appendList(outputFormat, "|,");
+
+            StringBuffer dirMask, fileMask;
+            getMasks(beforeWU, afterWU, dirMask, fileMask);
+            bool checkBeforeOrAfterWU = !isEmptyString(beforeWU) || !isEmptyString(afterWU);
+            bool sortInc = !isEmptyString(beforeWU) && isEmptyString(afterWU);
+
+            if (hasWUSOutput)
+                cmd->setWUSresult(WUSbuf);  // swap in/out (in case ever do multiple)
+
+            CIArrayOf<CWUData> wus;
+            unsigned wuCount = 0;
+            bool overflowed = false;
+            StringBuffer output;
+            Owned<IDirectoryIterator> dirIterator = dir->directoryFilesSorted(SD_bynameNC, sortInc, dirMask.isEmpty() ? nullptr : dirMask.str(), false, true);
+            ForEach(*dirIterator)
+            {
+                dirIterator->getName(output.clear());
+                if (!dirIterator->isDir() || !checkDirs(output.str()))
+                    continue;
+
+                Owned<IDirectoryIterator> fileIterator = dirIterator->query().directoryFilesSorted(SD_bynameNC, sortInc, fileMask.str(), false);
+                ForEach(*fileIterator)
+                {
+                    fileIterator->getName(output.clear());
+                    if (!parseAndCheckWUID(checkBeforeOrAfterWU, beforeWU, afterWU, output))
+                        continue;
+    
+                    const char *wuid = output.str();
```

Comment:
I think you've lost the check which skips workunits that are online ?
This is the old code:
```
                            if (isWild) {
                                if (!conn)
                                    conn.setown(querySDS().connect(baseXPath.str(), myProcessSession(), 0, 5*60*1000)); // connection to all
                            }
                            else {
                                VStringBuffer xpath("%s/%s", baseXPath.str(), mask);
                                conn.setown(querySDS().connect(xpath.str(), myProcessSession(), 0, 5*60*1000));
                            }
                            if ((isWild && !conn->queryRoot()->hasProp(wuid)) || (!isWild && !conn)) { // check not online
```

Reply:
@jakesmith I remove the check based on my understand to your previous comment: "Given the expense I think we may decided to return all, even if archived and in Dali." Please let me know if I mis-understand that comment.

Reply:
```
I wonder if it really needs to do this at all (check Dali).
Given the expense I think we may decided to return all, even if archived and in Dali.

The option (in Eclwatch) at the moment is 'Archive Only', but could be changed to a choice to include archive workunits in search instead of exclusively one or the other.
```

@wangkx - Beyond the scope of this PR, it would introduce a semantic change and would require UI changes too. 
This PR should maintain the same semantics that exist.

Reply:
I got it and will add it back.